packages/analytics-browser-test/test/web-attribution.test.ts

```diff
@@ -0,0 +1,603 @@
+              },
+            });
+            scope.done();
+            resolve();
+          }, 4000);
+        });
+      });
+
+      test('should start a new session with resetSessionCampaign enable and track campaign updates', async () => {
+        const url = 'https://www.example.com?utm_source=test_utm_source';
+        navigateTo(url);
+
+        let payload: any = undefined;
+        const scope = nock(httpEndPoint)
+          .post(path, (body: Record<string, any>) => {
+            payload = body;
+            return true;
+          })
+          .reply(200, success);
+
+        nock(httpEndPoint)
+          .post(path, (body: Record<string, any>) => {
+            payload = body;
+            return true;
+          })
+          .reply(200, success);
+
+        await client.init(apiKey, 'user1@amplitude.com', {
+          deviceId: UUID(),
+          defaultTracking: {
+            ...defaultTracking,
+            attribution: {
+              resetSessionOnNewCampaign: true,
+            },
+          },
+          sessionTimeout: 500,
+          flushIntervalMillis: 3000,
+        }).promise;
+
+        // refresh during the session.
+        const directUrl = 'https://www.example.com/?utm_content=test_utm_content';
+        navigateTo(directUrl);
+
+        // eslint-disable-next-line @typescript-eslint/no-misused-promises
+        setTimeout(async () => {
+          await client.init(apiKey, 'user1@amplitude.com', {
+            deviceId: UUID(),
+            defaultTracking: {
+              ...defaultTracking,
+              attribution: {
+                resetSessionOnNewCampaign: true,
+              },
+            },
+            sessionTimeout: 500,
+            flushIntervalMillis: 3000,
+          }).promise;
+        }, 100);
+
+        return new Promise<void>((resolve) => {
+          setTimeout(() => {
+            expect(payload).toEqual({
+              api_key: apiKey,
+              client_upload_time: event_upload_time,
+              events: [
+                generateAttributionEvent(++eventId, url),
+                generateSessionStartEvent(++eventId),
+                generatePageViewEvent(++eventId, 1, url),
+                generateSessionEndEvent(++eventId),
+                generateAttributionEvent(++eventId, directUrl),
+                generateSessionStartEvent(++eventId),
+                generatePageViewEvent(++eventId, 1, directUrl),
+              ],
+              options: {
+                min_id_length: undefined,
+              },
+            });
+            scope.done();
+            resolve();
+          }, 4000);
+        });
+      });
+    });
+  });
+
+  describe('process event', () => {
+    describe('in a new session', () => {
+      let eventId = -1;
+      beforeEach(() => {
+        eventId = -1;
+      });
+
+      afterEach(() => {
+        cleanup();
+      });
+
+      test('should track all UTMs and referrers while processing any event', async () => {
```

Comment:
This is a check for tracking UTMs and referrers while processing any event and a new session right? WIthin the session, the process function should not trigger any campaign identifies

Reply:
yes, if the event comes from a new session, it should track all campaigns.  if the event still within the session, it should not track anything. that has been covered at `Web attribution -> process event -> during a session -> should not track updated campaign `  https://github.com/amplitude/Amplitude-TypeScript/pull/730/files#diff-bbd8238204219a8329ef6d342f9447527516495d50bb268c6eb8ee14d42caa26R490

Reply:
ah missed the line 361 where it says `in a new session`